src/commands/push-to-app-collection.yaml

```diff
@@ -56,7 +56,21 @@ steps:
   - run: |
         cd .app-collection && git commit -m "$(cat ../.git_commit_message)"
   - run: |
-        cd .app-collection && git push || (echo "Error pushing app to the collection. See known errors in https://github.com/giantswarm/architect-orb/blob/master/README.md#push-to-app-collection" && exit 1)
+        cd .app-collection && \
+        declare -i attempt=0 && \
+        while ! git push; do
+          attempt+=1
+          if ((attempt >= 4)); then
+            printf '%s\n' \
+              "Error pushing app to the collection. See known errors in:" \
+              "https://github.com/giantswarm/architect-orb/blob/master/README.md#push-to-app-collection" \
+              "Giving up after $attempt failures." \
+              >&2
+            exit 1
+          fi
+          git pull --rebase --autostash
+          sleep 5
```

Comment:
What this does now is to pull and wait 5 seconds before pushing. This means we give 5 seconds more to screw it again. Let's flip it so that we wait 5 seconds and then pull and push immediately. This reduces the likelihood of getting screwed again.

Reply:
Good point.

Reply:
Done